source/common/http/conn_manager_impl.cc

```diff
@@ -115,7 +116,9 @@ ConnectionManagerImpl::ConnectionManagerImpl(ConnectionManagerConfig& config,
           overload_manager ? overload_manager->getThreadLocalOverloadState().getState(
                                  Server::OverloadActionNames::get().DisableHttpKeepAlive)
                            : Server::OverloadManager::getInactiveState()),
-      time_source_(time_source) {}
+      time_source_(time_source),
+      filter_state_(std::make_shared<StreamInfo::FilterStateImpl>(
```

Comment:
perf nit: is there a way that we can lazy initialize this? In many cases this will never be used.

Reply:
I made the change to do this by passing a reference of the ancestor shared_ptr to the FilterState.